actors/market/src/lib.rs

```diff
@@ -751,55 +767,28 @@ impl Actor {
                 let deal_ids = st.get_deals_for_epoch(rt.store(), i)?;
 
                 for deal_id in deal_ids {
-                    let deal = st.get_proposal(rt.store(), deal_id)?;
-                    let dcid = rt_deal_cid(rt, &deal)?;
-                    let state = st.find_deal_state(rt.store(), deal_id)?;
-
-                    // deal has been published but not activated yet -> terminate it
-                    // as it has timed out
-                    if state.is_none() {
-                        // Not yet appeared in proven sector; check for timeout.
-                        if curr_epoch < deal.start_epoch {
-                            return Err(actor_error!(
-                                illegal_state,
-                                "deal {} processed before start epoch {}",
-                                deal_id,
-                                deal.start_epoch
-                            ));
-                        }
-
-                        let slashed = st.process_deal_init_timed_out(rt.store(), &deal)?;
-                        if !slashed.is_zero() {
-                            amount_slashed += slashed;
-                        }
+                    let deal_proposal = match st.find_proposal(rt.store(), deal_id)? {
+                        Some(dp) => dp,
+                        None => continue,
```

Comment:
Please comment that a proposal might be missing having been cleaned up by manual settlement or termination prior to this schedule.